backend/battleship.py

```diff
@@ -0,0 +1,169 @@
+
+    def calculate_filled_cells(self):
+        """
+        Fill board with "ships".
+        """
+        for ship_size in [5, 4, 3, 3, 2]:
+            while True:
+                direction = random.choice(['horizontal', 'vertical'])
+                if direction == 'horizontal':
+                    row = random.randint(0, 9)
+                    col = random.randint(0, 10 - ship_size)  # Ship size
+                else:
+                    row = random.randint(0, 10 - ship_size)  # Ship size
+                    col = random.randint(0, 9)
+
+                # Check
+                if all(self.board[row + (direction == 'vertical') * i][col + (direction == 'horizontal') * i] == 0 for i in range(ship_size)):
+                    # Place ship and change cells
+                    for i in range(ship_size):
+                        self.board[row + (direction == 'vertical') * i][col + (direction == 'horizontal') * i] = 1
+                    break
+
+    @property
+    def timeout(self):
+        return self.timeout
+
+    @timeout.setter
+    def timeout(self, num_in_sec):
+        self.timeout = num_in_sec
+
+    def game_move_price(self, quantity):
+        self.game_move_price = quantity
+
+    @property
+    def reward_balance(self):
+        return self.reward_balance
+
+    @reward_balance.setter
+    def reward_balance(self, value):
+        self.reward_balance = value
+
+    def calculate_result(self, guess):
+        """
+        Find users move result
+        :param guess: int
+        :return: tuple
+        """
+        if 0 <= guess <= 99:
+            guess_row = guess // 10
+            guess_col = guess % 10
+
+            if self.board[guess_row][guess_col] == 1:  # Player won
+                logger.info(f'User hit the cell with reward. Coord: {guess}')
+                self.board[guess_row][guess_col] = 'X'
+                if all(cell == '-' for row in self.board for cell in row):
+                    logger.warn('Game should be reset as no cells')
+                return 1, "Win"
+            else:  # Player lose
+                logger.info(f'User hit empty cell. Coord: {guess}')
+                self.board[guess_row][guess_col] = '-'
+                if all(cell == '-' for row in self.board for cell in row):
+                    logger.warn('Game should be reset as no cells')
+                return 2, "Loss"
+
+    def call_move_result(self, move_id, game_id, result):
+        """
+        Calls to contract function moveResult with params:
+        :param move_id: int
+        :param game_id: int
+        :param result: int
+        :return:
+        """
+        transaction_hash = self.contract.functions.moveResult(
+            self.w3.to_wei(move_id, "wei"),  # Convert to type uint256
+            self.w3.to_wei(game_id, "wei"),  # Convert to type uint256
+            result[0]  # Status code
+        ).transact()
+        logger.info(f'SEND TRANSACTION WITH RESULT {result[1]} TO CONTRACT'
```

Comment:
Won't this log line appear only after tx result comes? Looks like it happens like this:
* send tx
* wait for tx
* compose the log line
* print it

Might be useful to have 2 lines:
* sending move result X
* tx details: YYY

Reply:
Ok, will fix in next commit.